src/components/common/ColonyTotalFunds/ColonyTotalFunds.tsx

```diff
@@ -82,24 +66,30 @@ const ColonyTotalFunds = () => {
 
   const isSupportedColonyVersion = true;
 
-  // if (!data || !currentToken || isLoadingTokenBalances) {
-  //   return (
-  //     <MiniSpinnerLoader
-  //       className={styles.main}
-  //       loadingText={MSG.loadingData}
-  //       titleTextValues={{ hasCounter: false }}
-  //     />
-  //   );
-  // }
+  const totalTokenBalance = useMemo(() => {
+    if (balances?.items && currentToken) {
+      return (
+        balances.items
+          /*
+           * If the domain is not set, then we're dealing with "All Domains" (id 0)
+           */
+          .filter((balance) => balance?.domain === null)
+          .find(
+            (balance) =>
+              balance?.token?.tokenAddress === currentToken.token.tokenAddress,
+          )
+      );
+    }
+    return { balance: '0' };
+  }, [balances, currentToken]);
 
   return (
     <div className={styles.main}>
       <div className={styles.selectedToken}>
         <Numeral
           className={styles.selectedTokenAmount}
           decimals={getTokenDecimalsWithFallback(currentToken?.token?.decimals)}
-          // value={currentToken.balances[COLONY_TOTAL_BALANCE_DOMAIN_ID].amount}
-          value={0}
+          value={totalTokenBalance?.balance as string}
```

Comment:
Perhaps leave a note here as to why we can type cast this safely (i.e. null domain is applied in the cache so we know `find` above will return)

Reply:
Did one better, I made a better fallback for `totalTokenBalance` removing the need to forcefully cast it